analysis/pre_analysis.py

```diff
@@ -0,0 +1,109 @@
+                test_name = "Anderson-Darling"
+
+            sns.boxplot(x=values, ax=axs[2], color='lightcoral')
+            axs[2].set_title(f'Boxplot of {column_name}')
+
+            plt.suptitle(
+                f'Distribution Analysis of {column_name}\nNormal Distribution: {normal} ({test_name} p = {p_value:.4f})')
+            plt.tight_layout()
+            plt.subplots_adjust(top=0.85)
+
+        else:
+            # Categorical distribution visualization
+            fig, ax = plt.subplots(1, 2, figsize=(18, 6))
+
+            # Define a consistent color palette for the unique categories
+            unique_categories = values.unique()
+            palette = sns.color_palette("viridis", len(unique_categories))
+            color_dict = dict(zip(unique_categories, palette))
+            order = sorted(unique_categories)  # Define order for consistent sorting
+
+            # Count plot for frequencies with hue and legend disabled
+            sns.countplot(x=values, ax=ax[0], hue=values, palette=color_dict, order=order, dodge=False, legend=False)
+            ax[0].set_title(f'Count of Categories in {column_name}')
+            ax[0].set_xlabel(column_name)
+            ax[0].set_ylabel('Count')
+
+            # Adding value labels on the count plot
+            for p in ax[0].patches:
+                ax[0].annotate(f'{int(p.get_height())}',
+                               (p.get_x() + p.get_width() / 2., p.get_height()),
+                               ha='center', va='baseline', fontsize=10, color='black', xytext=(0, 5),
+                               textcoords='offset points')
+
+            # Percentage plot with consistent colors, using `hue` to ensure order
+            value_counts = values.value_counts(normalize=True).reindex(order) * 100
+            sns.barplot(x=value_counts.index, y=value_counts.values, ax=ax[1], hue=value_counts.index,
+                        palette=color_dict, dodge=False, legend=False)
+            ax[1].set_title(f'Percentage Distribution of {column_name}')
+            ax[1].set_xlabel(column_name)
+            ax[1].set_ylabel('Percentage (%)')
+
+            # Adding percentage labels on the percentage plot
+            for p in ax[1].patches:
+                ax[1].annotate(f'{p.get_height():.1f}%',
+                               (p.get_x() + p.get_width() / 2., p.get_height()),
+                               ha='center', va='baseline', fontsize=10, color='black', xytext=(0, 5),
+                               textcoords='offset points')
+
+            plt.suptitle(f'Distribution Analysis of Categorical Column: {column_name}')
+            plt.tight_layout()
+
+        # Save or show plot
+        if save_to_file:
+            name = f"visualize_distribution_{column_name}.png"
+            plt.savefig(f"{save_to_file}/{name}")
+            print(f"Plot for '{column_name}' saved as {name} in {save_to_file}.")
+        else:
+            plt.show()
+        plt.close(fig)
```

Comment:
I will create method save_to_file(plt, pathname) in new file common.py and later you can use it here, but still merge it for now